packages/elements/src/types/elementDefinition.ts

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import type { ElementBox } from '@googleforcreators/units';
+
+/**
+ * Internal dependencies
+ */
+import type React from 'react';
+import type { Element } from './element';
+import type { ElementType } from './elementType';
+
+export interface EditProps<E extends Element> {
+  element: E;
+  box: ElementBox;
+  setLocalProperties: () => void;
+  getProxiedUrl: () => string;
+  updateElementById: () => void;
+  zIndexCanvas: number;
+  scaleMin: number;
+  scaleMax: number;
+}
+
+export interface FrameProps<E extends Element> {
+  element: E;
+}
+
+export interface OutputProps<E extends Element> {
+  element: E;
+  box: ElementBox;
+  flags: Record<string, boolean>;
+}
+
+export interface LayerIconProps<E extends Element> {
+  element: E;
+}
+
+export interface TextContentProps<E extends Element> {
+  element: E;
+  box: ElementBox;
+  previewMode?: boolean;
+  renderResourcePlaceholder?: boolean;
+}
+
+export interface DisplayProps<E extends Element> {
+  element: E;
+}
+
+export interface ElementDefinition<E extends Element = Element> {
+  type: ElementType;
+  name: string;
+  isMedia?: boolean;
+  getLayerText: (element: E) => string;
+  defaultAttributes: Partial<E>;
+  Edit: React.FC<EditProps<E>>;
```

Comment:
I've seen in multiple articles / resources mentioned that `React.FC` is better not to use, e.g. because it causes the component to implicitly accept children. I _think_ there were some other reasons as well. Thoughts? You probably have more information about it.

Reply:
That has been [changed in React 18](https://solverfox.dev/writing/no-implicit-children/). `React.FC` (or `React.FunctionComponent`) no longer implicitly accepts children.

**BUT** we're _not_ using React 18, and probably never will because of all the migration issues. So you're right, it should probably be changed to a more clean type.

Reply:
`React.VoidFunctionComponent` seems to be [the way](https://github.com/DefinitelyTyped/DefinitelyTyped/issues/34237#issuecomment-915178375) to do it pre-18. Thanks!

Reply:
Fixed in 1e3577e885.

Reply:
Nice! Should we add the eslint configuration as well to prefer this to `React.FC` mentioned in the [comment](https://github.com/DefinitelyTyped/DefinitelyTyped/issues/34237#issuecomment-915178375) you shared? cc @swissspidy

Reply:
Sure, why not, perhaps in a new PR?